addonscript/src/addonscript/script.py

```diff
@@ -64,6 +75,11 @@ async def async_main(context):
     async with aiohttp.ClientSession(connector=connector) as session:
         context.session = session
         build_locales_context(context)
+
+        # ensure the versions exists on AMO before proceeding with uploading langpacks
+        for locale in context.locales:
```

Comment:
We may want to build a set of versions, and only call the API once per version. I'm guessing that we'll have the exact same version for each xpi in a normal beta, so we might create the same version 5 times for l10n.

Reply:
Yeah, I was going to ask in the PR about this but you were faster :) I checked a couple of logs for a couple of releases and the version is actually the same for all. So yeah, I think we can still build the dict since that's an easy cheap operation and only perform the API call once.